Create CMIP7 based compsets and eam use_case files

wlin7 opened the pull request:
Create CMIP7-based compset using a slightly revised convention
`TIME[%scenario]_ATM[%phys]_LND[%phys...`
where `scenario` is related to forcing specifications.

The new compsets are in parallel to the CMIP6 based production compsets.
The compsets added are as follows,

- WCYCL1850-CMIP7
- WCYCL1950-CMIP7
- WCYCL20TR-CMIP7
- 2010-CMIP7

When the compset is used to create v3 high-resolution (ne120pg2) 
configuration, the high-resolution version (0.5 degree) of CMIP7
forcings will be used. Otherwise,  the 2-degree resolution version 
will be used for CMIP7 compsets.

1950, 2010, and 20TR compsets still use some CMIP6 based input files. 

CMIP7 based land surfdata (fsurdat) and lulc (flanduse_timeseries)  are 
currently available for 1850 and 20TR at r025 resolution. The original 
files are set as default. Simulations that intend to use modified fsurdat 
can override via user_nl_elm.

[BFB] No impact for previously existing compsets which use CMIP6

--------------------------------------------------------------------------
`e3sm_atm_integration` test suite passed in a test merge to master, except for `ERS_Ld5.ne4pg2_oQU480.F2010xx-ZM` which uses L72 instead of L128. (For updates to this branch up to 10/12/2025)

Data or specifications need to be updated.

GHGs: 
- Any revision for the prescribed concentrations for 1850, 1950, and 2010? (Yes, done)
- `Does 1pctCO2 file needs revision?` Supposed so, as the baseline (1850) value changes slightly.
- Historical transient CMIP7 GHGs data (updated)

Volcanic data
- File for the high-resolution version for the historical period is not finalized. 2deg version is used as a place holder
- `Need cmip7 data for 2010 compset`. Currently using the version from CMIP6.

E90, Linoz data, Chlorine loading, and oxidants
- Use CMIP6 based files extended using ssp245 (Chlorine loading and linoz data after 2017) 
- Same files for 1850 and 20TR compsets, but with different time-related data stream parameters, except E90 for the 2010 compset

2010
- Available `2010_clim` forcings only at low resolution (2 degree).
- `Solar input file still using the CMIP6 version`

Land CMIP7 forcing
- cmip7 popdens file goes via cmip7 use_case files (instead of namelist_defaults.xml)
- cmip7 fsurdat available at r025 grid. TBD if to further modify by zeroing out PCT_LAKE over the Antarctic and Greenland. If so,  for F20TR, also need to update the landuse_timeseries file.

hwangacme commented on components/eam/bld/namelist_files/use_cases/1850_eam_CMIP7_chemUCI-Linoz-mam5-vbs.xml:
```diff
@@ -0,0 +1,135 @@
+<sim_year>1850</sim_year>
+
+<!-- land datasets -->
+<!-- Set in components/clm/bld/namelist_files/use_cases/1850_CMIP6_control.xml -->
```
"CMIP6" here also require attention.

wlin7 replied:
Done.

hwangacme commented on components/elm/bld/namelist_files/use_cases/1850_CMIP7_control.xml:
```diff
@@ -0,0 +1,33 @@
+<stream_fldfilename_popdens phys="elm" use_cn=".true." >lnd/clm2/firedata/elmforc.popden_cmip7_gr_0.5x0.5_simyr1850-2025_c250822.nc</stream_fldfilename_popdens>
+
+
+<!-- CMIP6 DECK compsets -->
```
"CMIP6" wording and new land surface files required

hwangacme commented on components/elm/bld/namelist_files/use_cases/20thC_CMIP7_transient.xml:
```diff
@@ -0,0 +1,58 @@
+
+<sim_year>1850</sim_year>
+
+<sim_year_range>1850-2015</sim_year_range>
```
Will this year range be updated?

wlin7 replied:
Yes, changed to 1850-2023.

hwangacme commented on components/elm/bld/namelist_files/use_cases/20thC_CMIP7_transient.xml:
```diff
@@ -0,0 +1,58 @@
+
+<stream_fldfilename_popdens phys="elm" use_cn=".true." >lnd/clm2/firedata/elmforc.popden_cmip7_gr_0.5x0.5_simyr1850-2025_c250822.nc</stream_fldfilename_popdens>
+
+<!-- CMIP6 DECK compsets -->
```
The land input files names below will need to be updated to CMIP7

hwangacme commented on components/elm/cime_config/config_component.xml:
```diff
@@ -68,11 +68,14 @@
       <value compset="PDAY.*_ELM"            >1850-2100_rcp4.5_transient</value> 
       <!-- CMIP6 DECK compsets and related BGC compsets -->
       <value compset="1850(?:SOI)?_EAM%CMIP6_ELM">1850_CMIP6_control</value>
+      <value compset="1850(?:SOI)?.*CPLHIST_ELM">1850_CMIP6_control</value>
```
Not sure why this new compset is added for CMIP6.

hwangacme reviewed (commented):
I have finished reviewing the changed files and provided my comments.

hwangacme commented:
@wlin7 With the two new namelist variables added for the MAM5 CMIP7 tuning, the namelist_defaults_eam has changed. The two variables have a default value of 4.D-7 for the CMIP6 compsets, while their new value for CMIP7 is 1.D-6. Should this change be directly included in all the CMIP7 use_case files?

rljacob commented:
@xuezhengllnl please approve.

wlin7 commented on components/eam/bld/namelist_files/use_cases/1850_eam_CMIP7_chemUCI-Linoz-mam5-vbs-1pctCO2.xml:
```diff
@@ -0,0 +1,139 @@
+<cosp_lite>.true.</cosp_lite>
+
+<!-- Atmosphere initial condition -->
+<ncdata dyn="se" hgrid="ne30np4" nlev="72" >atm/cam/inic/homme/NGD_v3atm.ne30pg2.eam.i.0001-01-01-00000.c20230106.nc</ncdata>
```
This has been removed.

wlin7 commented on components/eam/bld/namelist_files/use_cases/1850_eam_CMIP7_chemUCI-Linoz-mam5-vbs-1pctCO2.xml:
```diff
@@ -0,0 +1,139 @@
+<?xml version="1.0"?>
+<namelist_defaults>
+
+<!-- Set default output options for CMIP6 simulations -->
```
Done

wlin7 commented:
Merged to next. The conflict was on extra (now removed) entries for idealized CO2 config, as follows, where CMIP7 entry is also added
```
     <value compset="_EAM%CMIP6.*"            >&eamv3_phys_defaults; &eamv3_chem_defaults;</value>
      <value compset="_EAM%CMIP6-1pctCO2" >&eamv3_phys_defaults; &eamv3_chem_defaults;</value>
      <value compset="_EAM%CMIP6-4xCO2"   >&eamv3_phys_defaults; &eamv3_chem_defaults;</value>
```
The extra lines appear to come from [a rebase](https://github.com/E3SM-Project/E3SM/commit/83efe4e107977a0e3849f470b8d909984585d7c9) for the GCAM-E3SM coupling branch
